.github/workflows/reusable_dockerfile_pipeline.yml

```diff
@@ -35,25 +36,53 @@ jobs:
-          # the inputs, if so, we change the package to it.
-          if [[ ${{ inputs.packageName }} != ${{ github.repository}} ]];then
-            # validate the input package name characters
-            if [[ ! "${{ inputs.packageName }}" =~ ^[A-Za-z0-9\-]+$ ]]; then
-              echo "------------------------------------------------------------"
-              echo "ERROR: Package name not valid! => [ ${{ inputs.packageName }} ]"
-              echo "ONLY can use: A-Za-z0-9\-"
-              echo "------------------------------------------------------------"
-              exit 1
-            fi
-            echo "IMAGE_NAME=$(echo ${{ github.repository_owner }}/${{ inputs.packageName }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
-            echo "image_name=$(echo ${{ github.repository_owner }}/${{ inputs.packageName }} | tr '[:upper:]' '[:lower:]')" >> "$GITHUB_OUTPUT"
-          fi
+          # Extract both the repository owner and repository name
+          # According to docs, github.repository is in the format of owner/repo, not just repo
+          # https://docs.github.com/en/actions/learn-github-actions/contexts#github-context
+            REPO_OWNER=$(echo "${{ github.repository }}" | awk -F'/' '{print $1}' | tr '[:upper:]' '[:lower:]')
+            REPO_NAME=$(echo "${{ github.repository }}" | awk -F'/' '{print $2}' | tr '[:upper:]' '[:lower:]')
+
+            echo "Repository Owner: $REPO_OWNER"
+            echo "Repository Name: $REPO_NAME"
+
+            echo "REPO_OWNER=$REPO_OWNER" >> $GITHUB_ENV
+            echo "repo_owner=$REPO_OWNER" >> "$GITHUB_OUTPUT"
+
+            echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
+            echo "short_sha=`echo ${GITHUB_SHA} | cut -c1-8`" >> "$GITHUB_OUTPUT"
+            # yamllint disable
+              echo "${{ inputs.packageName }}"
+
+              # If there is a user submitted package name, use it
+              if [[ -n "${{ inputs.packageName }}" ]]; then
+                PACKAGE_NAME=$(echo "${{ inputs.packageName }}" | tr '[:upper:]' '[:lower:]')
+                echo "INSIDOOOOOR: $PACKAGE_NAME"
+              elif [[ "${{ inputs.packageName }}" == "${{ github.repository }}" ]]; then
+                # If a user submitted package name that has the naming containing
+                # both the repository owner and repository name, we fail
+                # e.g: inputs.packageName = "celestiaorg/celestiaorg" is not allowed
+                echo "------------------------------------------------------------"
+                echo "ERROR: Package name not valid! => [ $PACKAGE_NAME} ]"
+                echo "Don't use the repository owner and repository name in the package name."
+                echo "------------------------------------------------------------"
+                exit 1
+              else
+                # Set the default package name to the repository name
+                PACKAGE_NAME=$REPO_NAME
+              fi
```

Comment:
This logic is flawed and doesn't match what we had in the previous PR. 
Because you are using if else if, you won't actually ever catch `"${{ inputs.packageName }}" == "${{ github.repository }}"` because in that case it is non-empty, which means the first if will be used. 

Please revert to the previous logic in #85 as that was cleaner and caught all edge cases.

Reply:
Done